pkg/autodiscovery/provider_descriptor.go

```diff
@@ -0,0 +1,127 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+package autodiscovery
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/providers"
+	"github.com/DataDog/datadog-agent/pkg/status/health"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// providerDescriptor keeps track of the configurations loaded by a certain
+// `ConfigProvider` and whether it should be polled or not.
+type providerDescriptor struct {
+	provider     providers.ConfigProvider
+	configs      []integration.Config
+	canPoll      bool
+	isPolling    bool
```

Comment:
I think we can get rid of this boolean and use only `canPoll`

Reply:
As the stop chan is not always created, I think I'd rather have an extra bool than creating and sending messages to a channel for nothing, WDYT?